edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -115,31 +102,67 @@ const LectureScreen: ApplicationRoute = () => {
 
     // Funtion to set Uri to the desired one from firebase storage
     async function getUri() {
-        setUri(await getDownloadURL(currentLecture.pdfUri));
-        console.log(uri);
+        const url = await getDownloadURL(currentLecture.pdfUri);
+        setUri(url);
     }
 
-    const renderQuestion = (questionKey: string) => (
+    // Function for updating the UI display values when switching through orientations an fullscreen modes
+    function updateUI() {
+        if (isFullscreen) {
+            setWidthPercent(["100%", "0%"]);
+            setHeightPercent(["100%", "0%"]);
+        } else if (isLandscape) {
+            setHeightPercent(["100%", "100%"]);
+            setWidthPercent(["60%", "40%"]);
+        } else {
+            setHeightPercent(["40%", "60%"]);
+            setWidthPercent(["100%", "100%"]);
+        }
+    }
+
+    // Portrait display for the screen
+    const setLandscape = async () => {
+        await ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.LANDSCAPE);
+        updateUI();
+    };
+
+    // Display each question given as parameters as a component 
+    const renderQuestion = (question: string) => (
```

Comment:
This should be extracted to its own component as it might be useful to be able to render a Question it outside of the scope of this function.

Reply:
Yes indeed I will work on it